tests/libwait/wait.go

```diff
@@ -104,27 +165,71 @@ func waitForVMIPhase(ctx context.Context, phases []v1.VirtualMachineInstancePhas
 	return vmi
 }
 
-func WaitForSuccessfulVMIStartIgnoreWarnings(vmi runtime.Object) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	wp := watcher.WarningsPolicy{FailOnWarnings: false}
-	return WaitForVMIStart(ctx, vmi, 180, wp)
+// WaitForVMIStartOrFailed blocks until the specified VirtualMachineInstance reaches either Failed or Running states
+func WaitForVMIStartOrFailed(vmi *v1.VirtualMachineInstance, seconds int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
+	return WaitForVMIPhase(vmi,
+		[]v1.VirtualMachineInstancePhase{v1.Running, v1.Failed},
```

Comment:
nit: readability wise I find the explicit slice declaration a bit clunky, maybe create a func that returns a slice of the type and call it 

    func UntilAnyPhase(phases ...v1.VirtualMachineInstancePhase) []v1.VirtualMachineInstancePhase {

Then this would look like 

```
return WaitForVMIPhase(vmi,
		UntilAnyPhase(v1.Running, v1.Failed),
		WithWarningsPolicy(&wp),
		WithTimeout(seconds),
		WithWaitForFail(true),
	)
```

It might even make sense to convert this into an `Option` that is required to provide, this way you could even have several "reached phase descriptions", i.e. even it should first reach this, and then that... But that is a speculation only.

WDYT?

Reply:
I know it does not look so nice, but introduce a function which simply returns the arguments looks weird too IMHO.
```
func UntilAnyPhase(phases ...v1.VirtualMachineInstancePhase) []v1.VirtualMachineInstancePhase {
    return phases
}
```
If you agree I will spend more time trying to find a better way with a follow up PR! :)

Reply:
SGTM, thanks!